### PR TITLE
[WIP] Avatar helper

### PR DESCRIPTION
**Pull request by lgavillet**
Add an avatar helper, to render avatars

### CODE REVIEW

**Comment by lwe** [comment]
Was thinking we could move use something like the following:

```haml
= mtl_avatar(user, options = {}) # => <span class="avatar"....>
= mtl_avatar_link(user, link, options = {}) # => <a class="avatar" href="...." ...

# both could delegate however to a low-level
= mtl_avatar_tag(name, fullname, image_url, options = {})
```

The higher-level methods would accept a user duck object that maybe responds to `.name` and `.avatar_url` or something?

/cc @sled

**Comment by lwe** [comment]
It could also be some sort of configuration, i.e. a proc, that can be configured to return the name and image, like:

```ruby
Mtl.avatar_name = ->(user) { "#{user.first_name} #{user.last_name}" }
Mtl.avatar_url = ->(user) { attachment_path(user.avatar) }
```

**Comment by lgavillet** [comment]
Then we don't cover the Bautro's external usecase, when a participant is only identified by an email. Then our initial method will be messed up or simply way too overkill. Or then we have to use the avatar_tag. Oh well, egal... Would be more in favor of a duck typing, rather than a 'Mtl.avatar_[xx]'.

**Comment by lwe** [comment]
Then let's start simple, lets use ducktyping

**Comment by lwe** [comment]
After discussions on Slack, lets go with the current version +/-, i.e. no ducktyping or anything, but I'd like to delegate the initials logic to this helper too, because we now have like 3-4 different implements in all the different projects... so I'd propose the following methods:

```
mtl_avatar(name, image_url = nil, options = {})
mtl_avatar_link(link, name, image_url = nil, options = {}) # not so happy here with the ordering, i.e. in regards to link_to helper, ideas?
```

Also default tag we maybe should use a `<span>`?

**Comment by lgavillet** [comment]
So then we support the email initials too?

File: spec/mtl/rails/avatar_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      expect(subject.mtl_avatar_initials('john@example.com')).to eq 'J'
+    end
+
+    it 'returns an empty string when nothins is provided' do
+      expect(subject.mtl_avatar_initials('')).to eq ''
+    end
+
+    it 'returns an empty string when a malformatted email is provided' do
+      expect(subject.mtl_avatar_initials('@example.com')).to eq ''
+    end
```
Comment (lwe) [inline spec/mtl/rails/avatar_helper_spec.rb]: add a spec for `nil`

File: lib/mtl/rails/avatar_helper.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      def mtl_avatar_link(url, name, image_url = nil, options = {})
+        options[:class] = ['avatar', options[:class]].compact
+        link_to (image_url.present? ? image_tag(image_url, alt: name) : '') + mtl_avatar_initials(name), url, options
+      end
+
+      # Renders an avatar, for the given name, with initials and an optional picture
+      #
+      # @param name [String]
+      # @param image_url [String] Optional image url for the avatar
+      def mtl_avatar(name, image_url = '')
```
Comment (lwe) [inline lib/mtl/rails/avatar_helper.rb]: shouldn't we also support passing `options` here, too, so one could add a color or something too?

File: lib/mtl/rails/avatar_helper.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    # ```
+    module AvatarHelper
+      # Renders an avatar link, for the given url with the name's initials
+      #
+      # @param url [String]
+      # @param name [String]
+      # @param image_url [String] Optional image url for the avatar
+      # @param options [Hash] Additional options that can be passed to `link_to`
+      def mtl_avatar_link(url, name, image_url = nil, options = {})
+        options[:class] = ['avatar', options[:class]].compact
```
Comment (lwe) [inline lib/mtl/rails/avatar_helper.rb]: would it make sense to also add `Mtl.effects` or is this overkill? 😄

File: README.md [hunk 4]
```diff
@@ -81,6 +81,14 @@ tags:
 = mtl_icon :send, size: 'large'
 ```
 
+**Avatars**, see [avatar_helper.rb][avatar_helper.rb] for details. Renders an avatar icon, with initials and
```
Comment (lwe) [inline README.md]: has been removed from README and moved to REFERENCE.md or the helper class docs

**Comment by lgavillet** [comment]
@marcopluess Can you review the added CSS, please? Thanks!

File: app/assets/stylesheets/mtl/extend/_avatars.scss [hunk 5]
```diff
@@ -1,24 +1,26 @@
...
 .mtl-layout-default .side-nav .avatar {
-  display: block;
-  width: 52px;
-  height: 52px;
-  line-height: 52px;
-  padding: 0;
-  border-radius: 26px;
+  display: inline-block;
+  border-radius: 50%;
+  height: 50px;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: @marcopluess what kind of avatar sizes do we actually need, also do we need additional classes like avatar-small or avatar-large?

File: app/assets/stylesheets/mtl/extend/_avatars.scss [hunk 6]
```diff
@@ -1,24 +1,26 @@
...
+  width: 50px;
+  font-family: Arial;
+  background: $grey;
+  overflow: hidden;
+  background-size: 50px 50px;
+  text-align: center;
+  line-height: 50px;
+  position: relative;
 
   //scss-lint:disable PlaceholderInExtend
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: I guess this is already part of `.scss-lint.yml` no?

File: app/assets/stylesheets/mtl/extend/_avatars.scss [hunk 7]
```diff
@@ -1,24 +1,26 @@
...
-  width: 52px;
-  height: 52px;
-  line-height: 52px;
-  padding: 0;
-  border-radius: 26px;
+  display: inline-block;
+  border-radius: 50%;
+  height: 50px;
+  width: 50px;
+  font-family: Arial;
```
Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: @marcopluess also we need to fix font-family, size and weight...

**Comment by lwe** [comment]
@marcopluess ping?

File: app/assets/stylesheets/mtl/extend/_avatars.scss (hunk 5 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: @lwe: found so far: 52x52, 32x32, 25x25... so yeah, we could go for .avatar-small (25), .avatar-large (52). what do you think?

Comment (lwe) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: so avatar-small, avatar-medium and avatar-large, do these correspond to the icon sizes in some way?

```html
      <!--
      Sizes:
      tiny: 1rem
      small: 2rem
      medium: 4rem
      large: 6rem
      -->
      <i class="large material-icons">insert_chart</i>
```
http://materializecss.com/icons.html

File: app/assets/stylesheets/mtl/extend/_avatars.scss (hunk 6 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: yes, at least in the mtl session branch

File: app/assets/stylesheets/mtl/extend/_avatars.scss (hunk 5 above)
Comment (marcopluess) [inline app/assets/stylesheets/mtl/extend/_avatars.scss]: implemented a proposal

**Pull request merged**